hcmanager/validation.go

```diff
@@ -0,0 +1,71 @@
+package hcmanager
+
+import (
+	hwcc "hardware-classification-controller/api/v1alpha1"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+)
+
+// ExtractAndValidateHardwareDetails this function will return map containing introspection details for a host.
+func (mgr HardwareClassificationManager) ExtractAndValidateHardwareDetails(extractedProfile hwcc.ExpectedHardwareConfiguration,
```

Comment:
Why is this function returning a mapping to an interface? We have a well-defined type that holds the same data, right?